sql/core/src/test/scala/org/apache/spark/sql/ShowCreateTableSuite.scala

```diff
@@ -226,6 +226,27 @@ abstract class ShowCreateTableSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("SPARK-37517: Keep consistent columns order with user specified") {
+    val t = "SPARK_37517"
+    withTable(t) {
+      sql(
+        s"""
+           |CREATE TABLE $t (
+           |  a bigint NOT NULL,
```

Comment:
I'm a little confused that the table will be created success although column `a` is nullable. It seems to me that partition columns should not be nullable. @cloud-fan